src/node/mod.rs

```diff
@@ -535,51 +532,32 @@ impl Node {
     // Message handling
     ////////////////////////////////////////////////////////////////////////////
 
-    fn try_handle_message(
-        &mut self,
-        sender: SocketAddr,
-        mut msg_with_bytes: MessageWithBytes,
-    ) -> Result<()> {
-        trace!("try handle message {:?}", msg_with_bytes);
+    fn try_handle_message(&mut self, sender: SocketAddr, msg: Message) -> Result<()> {
+        trace!("try handle message {:?}", msg);
 
-        self.try_relay_message(&msg_with_bytes)?;
+        self.try_relay_message(&msg)?;
```

Comment:
So we now lost the optimization we had where we only deserialize the `dst` first and only if the message is for us, we deserialize the rest of the message. Probably OK for now but we might want to add it back at some point, in some form.

Reply:
Yes, however now we don't pass on any unsigned message either and this is the trade-off. As we amplify messages per-hop then it may be best to stop invalid signed messages as close to the source as possible.  I think this also allows the sections to keep catching up with each other as we progress this area of the code.

Reply:
Ok, makes sense. Let's make things correct first and worry about optimizations later.